zetaclient/evm/outbounds.go

```diff
@@ -0,0 +1,243 @@
+	zetaTxHash, ballot, err := ob.zetaBridge.PostVoteOutbound(
+		cctxIndex,
+		receipt.TxHash.Hex(),
+		receipt.BlockNumber.Uint64(),
+		receipt.GasUsed,
+		transaction.GasPrice(),
+		transaction.Gas(),
+		receiveValue,
+		receiveStatus,
+		ob.chain,
+		nonce,
+		cointype,
+	)
+	if err != nil {
+		logger.Error().Err(err).Msgf("PostVoteOutbound: error posting vote for chain %d nonce %d outtx %s ", chainID, nonce, receipt.TxHash)
+	} else if zetaTxHash != "" {
+		logger.Info().Msgf("PostVoteOutbound: posted vote for chain %d nonce %d outtx %s vote %s ballot %s", chainID, nonce, receipt.TxHash, zetaTxHash, ballot)
+	}
+}
+
+// IsOutboundProcessed checks outtx status and returns (isIncluded, isConfirmed, error)
+// It also posts vote to zetacore if the tx is confirmed
+func (ob *ChainClient) IsOutboundProcessed(cctx *crosschaintypes.CrossChainTx, logger zerolog.Logger) (bool, bool, error) {
+	// skip if outtx is not confirmed
+	nonce := cctx.GetCurrentOutTxParam().OutboundTxTssNonce
+	if !ob.IsTxConfirmed(nonce) {
+		return false, false, nil
+	}
+	receipt, transaction := ob.GetTxNReceipt(nonce)
+	sendID := fmt.Sprintf("%d-%d", ob.chain.ChainId, nonce)
+	logger = logger.With().Str("sendID", sendID).Logger()
+
+	// get connector and erce20Custody contracts
+	connectorAddr, connector, err := ob.GetConnectorContract()
+	if err != nil {
+		return false, false, errors.Wrapf(err, "error getting zeta connector for chain %d", ob.chain.ChainId)
+	}
+	custodyAddr, custody, err := ob.GetERC20CustodyContract()
+	if err != nil {
+		return false, false, errors.Wrapf(err, "error getting erc20 custody for chain %d", ob.chain.ChainId)
+	}
+
+	// define a few common variables
+	var receiveValue *big.Int
+	var receiveStatus chains.ReceiveStatus
+	cointype := cctx.InboundTxParams.CoinType
+
+	// compliance check, special handling the cancelled cctx
+	if compliance.IsCctxRestricted(cctx) {
+		// use cctx's amount to bypass the amount check in zetacore
+		receiveValue = cctx.GetCurrentOutTxParam().Amount.BigInt()
+		receiveStatus := chains.ReceiveStatus_Failed
+		if receipt.Status == ethtypes.ReceiptStatusSuccessful {
+			receiveStatus = chains.ReceiveStatus_Success
+		}
+		ob.PostVoteOutbound(cctx.Index, receipt, transaction, receiveValue, receiveStatus, nonce, cointype, logger)
+		return true, true, nil
+	}
+
+	// parse the received value from the outtx receipt
+	receiveValue, receiveStatus, err = ParseOuttxReceivedValue(cctx, receipt, transaction, cointype, connectorAddr, connector, custodyAddr, custody)
+	if err != nil {
+		logger.Error().Err(err).Msgf("IsOutboundProcessed: error parsing outtx event for chain %d txhash %s", ob.chain.ChainId, receipt.TxHash)
+		return false, false, err
+	}
+
+	// post vote to zetacore
+	ob.PostVoteOutbound(cctx.Index, receipt, transaction, receiveValue, receiveStatus, nonce, cointype, logger)
+	return true, true, nil
+}
+
+// ParseAndCheckZetaEvent parses and checks ZetaReceived/ZetaReverted event from the outtx receipt
```

Comment:
Maybe some more comments on what is returned if certain events are found.

Reply:
see commit https://github.com/zeta-chain/node/pull/1989/commits/6c1685b677bc41243074daebfbd42de6485c752e